### PR TITLE
feat(first-severity-escalation): Add post process step for detecting new escalating issues

### LINKED ISSUE #60413: Create a new step in the post process pipeline to escalate new issues

### PR DESCRIPTION
**Pull request by wedamija**
This adds a `detect_new_escalation` step to post process group for errors. This is feature gated at the moment and so won't run for any customers.

Resolves https://github.com/getsentry/sentry/issues/60413

### CODE REVIEW

**Comment by wedamija** [comment]
Should update the substatus here too

File: src/sentry/tasks/post_process.py [hunk 1]
```diff
@@ -1322,11 +1325,53 @@ def should_postprocess_feedback(job: PostProcessJob) -> bool:
...
+    return 1000
+
+
+MAX_NEW_ESCALATION_AGE_HOURS = 24
+
+
+def detect_new_escalation(job: PostProcessJob):
+    """
+    Detects whether a new issue is escalating. New issues are issues less than
+    MAX_NEW_ESCALATION_AGE_HOURS hours old.
```
Comment (snigdhas) [inline src/sentry/tasks/post_process.py:1339]: Not particularly related to this PR, but we have a NEW substatus that means < 3 days old.

I'm wary about adding a second definition for 'new' especially related to escalating which is so closely tied to the issue states. Any way we can keep this from getting confusing down the line?

**Review by snigdhas — APPROVED** [review]
Curious about why you went for a project-scoped flag instead of organization?

**Comment by wedamija** [comment]
> Curious about why you went for a project-scoped flag instead of organization?

I was just following the pattern for other `first-severity` flags, I think we prefer to enable this per project.

**Pull request merged**